### PR TITLE
[PPP-3584] PRD - Security Vulnerability when using XML file as datasource

### PR DESCRIPTION
**Pull request by AliaksandrShuhayeu**
[PPP-3584] PRD - Security Vulnerability when using XML file as datasource

- disable default support DTD in pentaho report designer
- add property for enable/disable support DTD

### CODE REVIEW

File: engine/extensions-xpath/source/org/pentaho/reporting/engine/classic/extensions/datasources/xpath/LegacyXPathTableModel.java [hunk 1]
```diff
@@ -96,12 +104,20 @@ public LegacyXPathTableModel( final ResourceData xmlResource,
       columnTypes = new ArrayList<Class>();
       columnNames = new ArrayList<String>();
 
+      DocumentBuilderFactory df = DocumentBuilderFactory.newInstance();
+      final Configuration configuration = ClassicEngineBoot.getInstance().getGlobalConfig();
+      if ( !"true".equals( configuration
+              .getConfigProperty( "org.pentaho.reporting.designer.support.dtds.enable" ) ) ) {
```
Comment (tmorgner) [inline engine/extensions-xpath/source/org/pentaho/reporting/engine/classic/extensions/datasources/xpath/LegacyXPathTableModel.java]: Please change the property name to match the package of the XPAth datasource. It is not just report-designer but also BI-Server that are affected.

Use "org.pentaho.reporting.engine.classic.extensions.datasources.xpath.EnableDTDs" as property name, for instance.

**Comment by tmorgner** [comment]
Just reading the OWASP guide: I would also add "dbf.setXIncludeAware(false);" - or I craft you an XInclude that kills your server.

**Comment by AliaksandrShuhayeu** [comment]
@tmorgner I added dbf.setXIncludeAware(false); but it false by default and renamed property.

**Comment by pamval** [comment]
@tmorgner I'm reverting this change since we want to do a risk/scope assessment before including this in 7.0. This will be merged to master after we branch 7.0 tomorrow

**Comment by tmorgner** [comment]
Aye, no problem. There are other places that may need fixing as well.

**Pull request merged**